### PR TITLE
Many small clean-ups in the SSL/TLS code

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
We can make this code look a bit nicer.
Some of these things can be fixed because we now baseline on Java 11.

Modification:
Use modern Java features and fix warnings.

Result:
Simpler code.

### CODE REVIEW

File: handler/src/main/java/io/netty5/handler/ssl/PemReader.java [hunk 1]
```diff
@@ -95,7 +94,7 @@
             throw new CertificateException("found no certificates in input stream");
         }
 
-        return certs.toArray(new ByteBuf[0]);
+        return certs.toArray(ByteBuf[]::new);
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty5/handler/ssl/PemReader.java:97]: TIL about this syntax :)

File: handler/src/main/java/io/netty5/handler/ssl/SslUtils.java [hunk 2]
```diff
@@ -109,6 +109,7 @@
 
     private static final boolean TLSV1_3_JDK_SUPPORTED;
     private static final boolean TLSV1_3_JDK_DEFAULT_ENABLED;
+    public static final TrustManager[] EMPTY_TRUST_MANAGERS = new TrustManager[0];
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty5/handler/ssl/SslUtils.java:112]: nit:
```suggestion
   static final TrustManager[] EMPTY_TRUST_MANAGERS = new TrustManager[0];
```

File: handler/src/main/java/io/netty5/handler/ssl/PemReader.java (hunk 1 above)
Comment (chrisvest) [reply handler/src/main/java/io/netty5/handler/ssl/PemReader.java:97]: It's a method reference to the array constructor. 😉

**Review by normanmaurer — COMMENTED** [review]
LGTM... one nit

**Comment by chrisvest** [comment]
Not clear why one of the builds is marked as failed. In the log they look like everything succeeded.

**Pull request merged**